scaleway/resource_mnq_credential_test.go

```diff
@@ -0,0 +1,159 @@
+	mnq "github.com/scaleway/scaleway-sdk-go/api/mnq/v1alpha1"
+)
+
+func TestAccScalewayMNQCreeds_Basic(t *testing.T) {
+	tt := NewTestTools(t)
+	defer tt.Cleanup()
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:          func() { testAccPreCheck(t) },
+		ProviderFactories: tt.ProviderFactories,
+		CheckDestroy:      testAccCheckScalewayMNQnNamespaceCreedDestroy(tt),
+		Steps: []resource.TestStep{
+			{
+				Config: `
+					resource "scaleway_mnq_namespace" "main" {
+					  name     = "test-mnq-ns"
+					  protocol = "nats"
+					}
+			
+					resource "scaleway_mnq_credential" "main" {
+						name = "test-creed-ns"
+						namespace_id = scaleway_mnq_namespace.main.id
+					}
+				`,
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckScalewayMNQCreedExists(tt, "scaleway_mnq_credential.main"),
+					resource.TestCheckResourceAttr("scaleway_mnq_credential.main", "protocol", "nats"),
+					resource.TestCheckResourceAttrSet("scaleway_mnq_credential.main", "nats_credentials.0.content"),
+					resource.TestCheckResourceAttrSet("scaleway_mnq_namespace.main", "region"),
+				),
+			},
+			{
+				Config: `
+					resource "scaleway_mnq_namespace" "main" {
+					  name     = "test-mnq-sqs"
+					  protocol = "sqs_sns"
+					}
+					
+					resource "scaleway_mnq_credential" "main" {
+					  name         = "test-creed-sqs"
+					  namespace_id = scaleway_mnq_namespace.main.id
+					  sqs_sns_credentials {
+						permissions {
+						  can_publish = true
+						  can_receive = true
+						  can_manage  = true
+						}
+					  }
+					}
+				`,
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckScalewayMNQCreedExists(tt, "scaleway_mnq_credential.main"),
+					resource.TestCheckResourceAttr("scaleway_mnq_credential.main", "name", "test-creed-sqs"),
+					resource.TestCheckResourceAttr("scaleway_mnq_credential.main", "protocol", "sqs_sns"),
+					resource.TestCheckResourceAttrSet("scaleway_mnq_credential.main", "sqs_sns_credentials.0.permissions.0.can_publish"),
+					resource.TestCheckResourceAttrSet("scaleway_mnq_credential.main", "sqs_sns_credentials.0.permissions.0.can_manage"),
+					resource.TestCheckResourceAttrSet("scaleway_mnq_credential.main", "sqs_sns_credentials.0.permissions.0.can_receive"),
+					resource.TestCheckResourceAttrSet("scaleway_mnq_credential.main", "sqs_sns_credentials.0.secret_key"),
+					resource.TestCheckResourceAttrSet("scaleway_mnq_credential.main", "sqs_sns_credentials.0.access_key"),
+					resource.TestCheckResourceAttrSet("scaleway_mnq_namespace.main", "region"),
+				),
+			},
+			{
+				Config: `
+					resource "scaleway_mnq_namespace" "main" {
+					  name     = "test-mnq-sqs-update"
+					  protocol = "sqs_sns"
+					}
+					
+					resource "scaleway_mnq_credential" "main" {
+					  name         = "test-mnq-sqs-update"
+					  namespace_id = scaleway_mnq_namespace.main.id
+					  sqs_sns_credentials {
+						permissions {
+						  can_publish = true
```

Comment:
Could you test it works by changing the permissions?

Reply:
test setup with permissions